From 134f532c67b5281f095e2fbc51eb26fb7f86c1db Mon Sep 17 00:00:00 2001
From: Zhao Jiazhong <zhaojiazhong-hf@loongson.cn>
Date: Mon, 10 Oct 2022 13:22:35 +0000
Subject: [PATCH] Bug 1792981 - [loong64] Enable JIT compiler of loong64 port
 by default. r=jandem

Fix a build with JIT issue on native loongarch64 machine, and enable JIT by default.

Differential Revision: https://phabricator.services.mozilla.com/D158397
---
 js/moz.configure                   |  2 ++
 js/src/wasm/WasmSignalHandlers.cpp | 10 +++++-----
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/js/moz.configure b/js/moz.configure
index 7a241cac76..2bb7979e36 100644
--- a/js/moz.configure
+++ b/js/moz.configure
@@ -234,6 +234,8 @@ def jit_codegen(jit_enabled, simulator, target):
         return namespace(arm64=True)
     elif target.cpu == "x86_64":
         return namespace(x64=True)
+    elif target.cpu == "loongarch64":
+        return namespace(loong64=True)
 
     return namespace(**{str(target.cpu): True})
 
diff --git a/js/src/wasm/WasmSignalHandlers.cpp b/js/src/wasm/WasmSignalHandlers.cpp
index 4a45905431..074c373106 100644
--- a/js/src/wasm/WasmSignalHandlers.cpp
+++ b/js/src/wasm/WasmSignalHandlers.cpp
@@ -158,10 +158,10 @@ using mozilla::DebugOnly;
 #      define R32_sig(p) ((p)->uc_mcontext.gp_regs[32])
 #    endif
 #    if defined(__linux__) && defined(__loongarch__)
-#      define EPC_sig(p) ((p)->uc_mcontext.pc)
-#      define RRA_sig(p) ((p)->uc_mcontext.gregs[1])
-#      define RSP_sig(p) ((p)->uc_mcontext.gregs[3])
-#      define RFP_sig(p) ((p)->uc_mcontext.gregs[22])
+#      define EPC_sig(p) ((p)->uc_mcontext.__pc)
+#      define RRA_sig(p) ((p)->uc_mcontext.__gregs[1])
+#      define R03_sig(p) ((p)->uc_mcontext.__gregs[3])
+#      define RFP_sig(p) ((p)->uc_mcontext.__gregs[22])
 #    endif
 #  elif defined(__NetBSD__)
 #    define EIP_sig(p) ((p)->uc_mcontext.__gregs[_REG_EIP])
@@ -403,7 +403,7 @@ struct macos_aarch64_context {
 #  elif defined(__loongarch__)
 #    define PC_sig(p) EPC_sig(p)
 #    define FP_sig(p) RFP_sig(p)
-#    define SP_sig(p) RSP_sig(p)
+#    define SP_sig(p) R03_sig(p)
 #    define LR_sig(p) RRA_sig(p)
 #  endif
 
-- 
2.45.0

